### PR TITLE
SLING-10586 prevent java.nio.file.FileSystemNotFoundException during

### PR DESCRIPTION
**Pull request by kwin**
DefaultArtifactHandler.getArtifact(...)

### CODE REVIEW

**Comment by kwin** [comment]
@karlpauls Can you provide a test case?

File: src/main/java/org/apache/sling/feature/io/artifacts/ArtifactManager.java [hunk 1]
```diff
@@ -383,16 +385,17 @@ public URL getArtifact(final String url, final String relativeCachePath) {
             logger.debug("Checking url to be local file {}", url);
             // check if this is already a local file
             try {
-                final Path f = Paths.get(new URL(url).toURI());
-                if (Files.exists(f)) {
-                    this.config.incLocalArtifacts();
-                    return f.toUri().toURL();
+                URI uri = new URI(url);
+                if (FileSystems.getDefault().provider().getScheme().equals(uri.getScheme())) {
```
Comment (kwin) [inline src/main/java/org/apache/sling/feature/io/artifacts/ArtifactManager.java:389]: Do we need to support URIs without a scheme for local files?

Comment (karlpauls) [reply src/main/java/org/apache/sling/feature/io/artifacts/ArtifactManager.java:389]: I think that is handled some place else

**Review by karlpauls — APPROVED** [review]
I added a test case that should capture the different scenarios.

**Pull request merged**